File: Strategy/Java/bank/Account.java
package bank;

public abstract class Account
{
	private int balance;
	
	private WithdrawStrategy withdrawBehavior;
	private CalculateInterestStrategy calculateInterestBehavior;
	
	public Account(WithdrawStrategy ws, CalculateInterestStrategy cs)
	{
		balance = 0;
		withdrawBehavior = ws;
		calculateInterestBehavior = cs;
	}
	
	public void setBalance(int balance)
	{
		this.balance = balance;
	}
	
	public int getBalance()
	{
		return balance;
	}

	public void deposit(int money)
	{
		balance += money;
		System.out.println("Deposit Money : " + money + ", Balance : " + balance);
	}
	
	public void withdraw(int money)
	{
		withdrawBehavior.withdraw(this, money);
	}
	
	public void calculateInterest()
	{
		int interest = calculateInterestBehavior.calculateInterest(this);
		System.out.println("Interest : " + interest);
	}
}

File: Strategy/C++/Account.cpp
#include "Account.h"


Account::Account()
{

}

Account::Account(WithdrawStrategy* ws, CalculateInterestStrategy* cs)
{
	withdrawBehavior = ws;
	calculateInterestBehavior = cs;
}

Account::~Account()
{
}

int Account::getBalance()
{
	return balance;
}

void Account::deposit(int money)
{
	balance += money;
	cout << "Deposit Money : " << money << ", Balance : " << balance << endl;
}

void Account::withdraw(int money)
{
	balance = withdrawBehavior->withdraw(this, money);
}

void Account::calculateInterest()
{
	int interest = calculateInterestBehavior->calculateInterest(this);
	cout << "Interest : " << interest << endl;
}
